core/src/hash/accumulator.rs

```diff
@@ -103,6 +120,7 @@ impl<'a> Accumulator for Wrap<'a> {
     }
 
     fn write(&mut self, bytes: &[u8]) {
+        assert!(self.len() + bytes.len() <= self.capacity());
```

Comment:
Isn't that redundant? Your slicing operation will panic anyways when this happens.

Reply:
true